### PR TITLE
(data-topcitation) Ajout du WS asynchrone sur les références phares

### PR DESCRIPTION
**Pull request by revolj**

### CODE REVIEW

File: services/data-topcitation/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    charset-normalizer==3.3.2 \
+    idna==3.7 \
+    requests==2.32.3 \
+    urllib3==2.2.2
+
+# Install all node dependencies
+RUN npm install \
+    @ezs/teeft@2.3.1 \
+    @ezs/strings@1.0.3
+
```
Comment (parmentf) [inline services/data-topcitation/Dockerfile:18]: **suggestion:** ‏If you don't use them, don't install them

It think it's part of the template, to show how to install npm packages, that's all.

```suggestion
```

File: services/data-topcitation/config.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "environnement": {
+        "EZS_TITLE": "Référence phare d'un corpus",
+        "EZS_DESCRIPTION": "à partir d'une information (pour l'instant doi) récupère les citatio",
```
Comment (parmentf) [inline services/data-topcitation/config.json:4]: **typo:** ‏
```suggestion
        "EZS_DESCRIPTION": "À partir d'une information (pour l'instant doi) récupère les citations",
```

File: services/data-topcitation/swagger.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            "name": "Inist-CNRS",
+            "url": "https://www.inist.fr/nous-contacter/"
+        }
+    },
+    "servers": [
+        {
+            "x-comment": "Will be automatically completed by the ezs server."
+        },
+        {
+            "url": "http://vptdmservices.intra.inist.fr:49225/",
```
Comment (parmentf) [inline services/data-topcitation/swagger.json:18]: **note (non-blocking):** ‏The final machine will be `vptdmjobs`, don't forget it when deploying!

File: services/data-topcitation/swagger.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            "description": "Latest version for production",
+            "#DISABLED#x-profil": "Standard"
```
Comment (parmentf) [inline services/data-topcitation/swagger.json:20]: **note (non-blocking):** At deployment, don't forget to activate `x-profil`.‏

File: services/data-topcitation/v1/retrieve-json.ini [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+post.responses.default.content.application/json.example.5.key = n: precipitation
+post.responses.default.content.application/json.example.5.freq = 7
+post.responses.default.content.application/json.example.6.key = n: genus
+post.responses.default.content.application/json.example.6.freq = 7
+post.responses.default.content.application/json.example.7.key = n: obliquiloculata
+post.responses.default.content.application/json.example.7.freq = 7
+post.responses.default.content.application/json.example.8.key = n: telescopus
+post.responses.default.content.application/json.example.8.freq = 6
+post.responses.default.content.application/json.example.9.key = nn: clay mineral
+post.responses.default.content.application/json.example.9.freq = 6
```
Comment (parmentf) [inline services/data-topcitation/v1/retrieve-json.ini:36]: **polish:** ‏Maybe you could put here an example matching your service.

**Review by parmentf — CHANGES_REQUESTED** [review]
Write a `tests.hurl`, you can use `data-termsuite` as an example. Two queries:

1. the call to your main route (`v1/topcitation`)
2. the call to `v1/retrieve-json` with a `delay` option, and the variable containing the value returned by the first route.

File: services/data-topcitation/tests.hurl [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+# That query gives the process identifier.
+# Next, you can call the retrieve URL by putting the identifier in the value field.
+
+# Version 4.1.0 of hurl added a delay option, which value is milliseconds.
+# https://hurl.dev/blog/2023/09/24/announcing-hurl-4.1.0.html#add-delay-between-requests
+
+POST {{host}}/v1/retrieve-json?indent=true
+content-type: application/json
+[Options]
+delay: 10000
```
Comment (leogail) [inline services/data-topcitation/tests.hurl:37]: **praise:** Si ça bug, ici il faut augmenter le délai (à 10 secondes). Test avec un temps plus élevé

**Review by leogail — CHANGES_REQUESTED** [review]

**Review by parmentf — APPROVED** [review]

**Pull request merged**